NPE in AbstractToolsMojo.initializeProperties when enabling bindSessionUserOverrideProperties

To access maven property values that were overridden on the command line I enabled:
```xml
<bindAllProjectProperties>true</bindAllProjectProperties>
<bindSessionUserOverrideProperties>true</bindSessionUserOverrideProperties>
```

The moment I do this and run my build I get the following exception:
```python
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal org.codehaus.gmavenplus:gmavenplus-plugin:1.13.0:execute (compile) on project jstuff-parent: Execution compile of goal org.codehaus.gmavenplus:gmavenplus-plugin:1.13.0:execute failed.            
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:215)                                                                                                                                                                                           
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)                                                                                                                                                                                           
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)                                                                                                                                                                                           
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)                                                                                                                                                                  
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)                                                                                                                                                                   
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)                                                                                                                                                     
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)                                                                                                                                                                                   
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)                                                                                                                                                                                                            
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)                                                                                                                                                                                                            
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)                                                                                                                                                                                                              
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:972)                                                                                                                                                                                                                  
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:293)                                                                                                                                                                                                                   
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:196)                                                                                                                                                                                                                     
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)                                                                                                                                                                                                      
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)                                                                                                                                                                                    
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)                                                                                                                                                                            
    at java.lang.reflect.Method.invoke (Method.java:566)                                                                                                                                                                                                                          
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)                                                                                                                                                                                       
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)                                                                                                                                                                                               
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)                                                                                                                                                                                     
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)                                                                                                                                                                                                 
Caused by: org.apache.maven.plugin.PluginExecutionException: Execution compile of goal org.codehaus.gmavenplus:gmavenplus-plugin:1.13.0:execute failed.                                                                                                                           
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:148)                                                                                                                                                                         
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:210)                                                                                                                                                                                           
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)                                                                                                                                                                                           
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)                                                                                                                                                                                           
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)                                                                                                                                                                  
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)                                                                                                                                                                   
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)                                                                                                                                                     
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)                                                                                                                                                                                   
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)                                                                                                                                                                                                            
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)                                                                                                                                                                                                            
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)                                                                                                                                                                                                              
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:972)                                                                                                                                                                                                                  
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:293)                                                                                                                                                                                                                   
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:196)                                                                                                                                                                                                                     
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)                                                                                                                                                                                                      
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)                                                                                                                                                                                    
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)                                                                                                                                                                            
    at java.lang.reflect.Method.invoke (Method.java:566)                                                                                                                                                                                                                          
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)                                                                                                                                                                                       
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)                                                                                                                                                                                               
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)                                                                                                                                                                                     
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)                                                                                                                                                                                                 
Caused by: java.lang.NullPointerException                                                                                                                                                                                                                                         
    at java.util.concurrent.ConcurrentHashMap.putVal (ConcurrentHashMap.java:1011)                                                                                                                                                                                                
    at java.util.concurrent.ConcurrentHashMap.put (ConcurrentHashMap.java:1006)                                                                                                                                                                                                   
    at java.util.Properties.put (Properties.java:1340)                                                                                                                                                                                                                            
    at org.codehaus.gmavenplus.mojo.AbstractToolsMojo.initializeProperties (AbstractToolsMojo.java:188)                                                                                                                                                                           
    at org.codehaus.gmavenplus.mojo.ExecuteMojo.setupShell (ExecuteMojo.java:181)                                                                                                                                                                                                 
    at org.codehaus.gmavenplus.mojo.ExecuteMojo.doExecute (ExecuteMojo.java:139)                                                                                                                                                                                                  
    at org.codehaus.gmavenplus.mojo.ExecuteMojo.execute (ExecuteMojo.java:90)                                                                                                                                                                                                     
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)                                                                                                                                                                         
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:210)                                                                                                                                                                                           
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)                                                                                                                                                                                           
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)                                                                                                                                                                                           
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)                                                                                                                                                                  
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)                                                                                                                                                                   
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)                                                                                                                                                     
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)                                                                                                                                                                                   
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)                                                                                                                                                                                                            
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)                                                                                                                                                                                                            
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)                                                                                                                                                                                                              
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:972)                                                                                                                                                                                                                  
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:293)                                                                                                                                                                                                                   
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:196)                                                                                                                                                                                                                     
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)                                                                                                                                                                                                      
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)                                                                                                                                                                                    
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)                                                                                                                                                                            
    at java.lang.reflect.Method.invoke (Method.java:566)                                                                                                                                                                                                                          
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)                                                                                                                                                                                       
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)                                                                                                                                                                                               
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)                                                                                                                                                                                     
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)                                                                                                                                                                                                 
```

When I set `bindSessionUserOverrideProperties` to `false` the NPE does not occur.

--- a/pom.xml
+++ b/pom.xml
@@ -5,7 +5,7 @@
   <groupId>org.codehaus.gmavenplus</groupId>
   <artifactId>gmavenplus-plugin</artifactId>
   <packaging>maven-plugin</packaging>
-  <version>1.14.0-SNAPSHOT</version>
+  <version>1.13.1-SNAPSHOT</version>
 
   <properties>
     <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>

--- a/src/main/java/org/codehaus/gmavenplus/mojo/AbstractToolsMojo.java
+++ b/src/main/java/org/codehaus/gmavenplus/mojo/AbstractToolsMojo.java
@@ -185,7 +185,9 @@ public abstract class AbstractToolsMojo extends AbstractGroovyMojo {
                 properties.putAll(session.getUserProperties());
             } else if (bindAllProjectProperties && bindSessionUserOverrideProperties && project != null) {
                 for (Object key : project.getProperties().keySet()) {
-                    properties.put(key, session.getUserProperties().get(key));
+                    if (session.getUserProperties().get(key) != null) {
+                        properties.put(key, session.getUserProperties().get(key));
+                    }
                 }
             }
         }

--- a/src/test/java/org/codehaus/gmavenplus/mojo/AbstractToolsMojoTest.java
+++ b/src/test/java/org/codehaus/gmavenplus/mojo/AbstractToolsMojoTest.java
@@ -143,6 +143,20 @@ public class AbstractToolsMojoTest {
     }
 
     @Test
+    public void testBindAllProjectPropertiesAndBindSessionUserOverridePropertiesWhenEmpty() {
+        MavenSession session = mock(MavenSession.class);
+        Properties projectProperties = new Properties();
+        projectProperties.put("foo", "bar");
+        doReturn(new Properties()).when(session).getUserProperties();
+        doReturn(projectProperties).when(project).getProperties();
+        testMojo.session = session;
+        testMojo.bindAllProjectProperties = true;
+        testMojo.bindSessionUserOverrideProperties = true;
+
+        testMojo.initializeProperties();
+    }
+
+    @Test
     public void testSessionPropertiesOverrideProjectPropertiesAndIncludesOthers() {
         Properties projectProperties = new Properties();
         projectProperties.put("foo", "bar");

